From b4c5b43ecd749fbb7da035bf5a80f5373b439953 Mon Sep 17 00:00:00 2001
From: Daniel Templeton <templedf@apache.org>
Date: Tue, 23 May 2017 15:33:06 -0700
Subject: [PATCH 2456/2848] YARN-6050 Addendum. AMs can't be scheduled on
 racks or nodes (rkanter)

Added missing code that breaks node labels.

Change-Id: I9fe8876553363e3372218ee89f797132290f05b5
---
 .../yarn/server/resourcemanager/RMAppManager.java  |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/RMAppManager.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/RMAppManager.java
index 1975127..b473621 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/RMAppManager.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/RMAppManager.java
@@ -401,6 +401,12 @@ private RMAppImpl createAndPopulateNewRMApp(
           amReq.setPriority(RMAppAttemptImpl.AM_CONTAINER_PRIORITY);
         }
 
+        // set label expression for AM ANY request if not set
+        if (null == anyReq.getNodeLabelExpression()) {
+          anyReq.setNodeLabelExpression(submissionContext
+              .getNodeLabelExpression());
+        }
+
         // Put ANY request at the front
         if (!amReqs.get(0).equals(anyReq)) {
           amReqs.remove(anyReq);
-- 
1.7.9.5

